### ISSUE TITLE
tracer: What should a user do about the "applications" list?

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#15365]
For the leftover software list, aka: "applications", there's no clear action on what a user should do to resolve it.

(Following up from the thread that starts @ https://github.com/cockpit-project/cockpit/pull/14581#issuecomment-780201745):

We should and should do more. Here are a few ideas of things we may want to consider:

1. A button to mark as resolved.
2. Show PIDs.
3. Handle systemd-related things in a more elegant way?
4. Have some explanatory help?
5. Special-case Cockpit-specific things?

**Comment by garrett** [comment]
Here's a list of software that required things to happen, as discovered by tracer and displayed in Cockpit:
![tracer list](https://user-images.githubusercontent.com/10246/108095637-5dfe3e00-7080-11eb-9053-31f0f355aec7.png)

So we have:

- cockpit-bridge
- cockpit-pcp
- sudo
- dnsmasq
- bash
- (sd-parm)
- cockpit-session

What can we do for each of this? Some would require a Cockpit restart, right?

How should sd-parm be handled?

How should sudo be handled? It's probably there due to Cockpit, right?

Bash is tricky. Can we tell how it's used? Is someone logged in? Is it from a shell script?

What's up with dnsmasq? That's due to NetworkManager, I guess?

How many of these will often show up in a Cockpit session?

**Comment by garrett** [comment]
My first guess is that we probably want to have a way to get details about these items. It should probably have some contextual actions.

So, here's _a really bad_ example of a dialog with all sorts of issues, but it's a quick set of ideas that might help the discussion along:

![software-outdated-2021-02-17-1853](https://user-images.githubusercontent.com/10246/108247616-2dd39f80-7153-11eb-9ba8-1525a7d31935.png)

**Comment by garrett** [comment]
Oh, forgot to tag @skobyda, @martinpitt... as we're continuing the discussion here. :wink:

**Comment by martinpitt** [comment]
Keeping this open, as it is still current in the sense of "nothing changed here". However, this is very far into the area of diminishing returns. Building an understanding of non-service processes into Cockpit, offering something sensible and reliable for these *with confidence* is an enormous task. Frankly, as a user if you want to make sure your security updates apply, just reboot the machine. The one thing that I veto is to add a [Kill] button next to these processes -- because people will use it then.

In other words, this is a classic case of "wontfix, but we don't want to admit it"..
